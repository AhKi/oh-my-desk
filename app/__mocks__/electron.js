export default jest.fn();

export const webContents = {
  getAllWebContents: jest.fn(() => []),
};

export const ipcMain = {
  on: jest.fn(),
};

export const BrowserWindow = jest.fn(info => ({
  close: jest.fn(),
  loadURL: jest.fn(),
  once: jest.fn(),
  on: jest.fn(),
  setAlwaysOnTop: jest.fn(),
  show: jest.fn(),
  showInactive: jest.fn(),
  getContentBounds: jest.fn(() => ({
    x: info.x,
    y: info.y,
    height: info.height,
    width: info.width,
  })),
  webContents: {
    on: jest.fn(),
    send: jest.fn(),
    openDevTools: jest.fn(),
  },
}));

export const ipcRenderer = {
  on: jest.fn(),
  send: jest.fn(),
};

export const remote = {
  getGlobal: jest.fn(),
  getCurrentWindow: jest.fn(),
  Menu: {
    buildFromTemplate: jest.fn(() => ({
      popup: jest.fn(),
    })),
  },
  app: {
    getVersion: jest.fn(),
  },
};

export const shell = {
  openExternal: jest.fn(),
};

export const app = {
  getName: jest.fn(() => 'oh-my-desk'),
  getPath: jest.fn(arg => arg),
  getVersion: jest.fn(),
  getAppPath: jest.fn(() => 'mock-path'),
  isReady: jest.fn(),
  on: jest.fn(),
};

export const Menu = {
  buildFromTemplate: jest.fn(() => ({
    popup: jest.fn(),
  })),
  setApplicationMenu: jest.fn(),
};

export const globalShortcut = {
  register: jest.fn(),
  unregister: jest.fn(),
};
