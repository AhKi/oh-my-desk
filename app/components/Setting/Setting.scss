.Setting {
  width: 100%;
  height: 100%;
}

.Setting__content {
  margin: 20px;
  padding: 10px;
  background-color: #f2f2f2;
  border: 1px solid #e1e1e1;
  border-radius: 4px;
}
