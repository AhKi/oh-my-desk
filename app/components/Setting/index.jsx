import React from 'react';

class Setting extends React.Component {
	render() {
		return (
			<div>
				Setting
			</div>
		);
	}
}

export default Setting;