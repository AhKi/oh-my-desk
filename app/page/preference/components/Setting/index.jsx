import React from 'react';

const propTypes = {};
const defaultProps = {};

class Setting extends React.Component {
  render() {
    return (
      <div>Setting</div>
    );
  }
}

Setting.propTypes = propTypes;
Setting.defaultProps = defaultProps;

export default Setting;
