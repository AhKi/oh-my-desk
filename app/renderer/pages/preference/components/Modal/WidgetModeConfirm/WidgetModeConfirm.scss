.WidgetModeConfirm__container {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  padding: 20px;
  font-family: inherit;
}

.WidgetModeConfirm__content {
  padding: 10px 0;
  font-size: 16px;
}

.WidgetModeConfirm__btn-set {
  display: flex;
  justify-content: center;
  margin: 10px;
}
