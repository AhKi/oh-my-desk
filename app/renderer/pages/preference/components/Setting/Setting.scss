.Setting {
  margin: 0 15px 15px 15px;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.Setting__title {
  font-weight: 600;
  font-size: 14px;
  margin: 10px 0;
}

.Setting__list {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px;
  background: white;
  font-size: 14px;
  border-bottom: 1px solid $color-cool-gray200;
  height: 45px;
}

.Setting__list-set {
  box-shadow: 1px 2px rgba(0, 0, 0, 0.2);
}

.Setting__select {
  width: 400px;
}
