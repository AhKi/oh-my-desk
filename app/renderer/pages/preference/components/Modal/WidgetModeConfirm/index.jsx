import React from 'react';
import PropTypes from 'prop-types';
import i18n from 'constants/i18n';
import './WidgetModeConfirm.scss';

const propTypes = {
  onChangeMode: PropTypes.func,
  onModalClose: PropTypes.func,
};
const defaultProps = {
  onChangeMode() {},
  onModalClose() {},
};

class WidgetModeConfirm extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange() {
    const { onChangeMode, onModalClose } = this.props;

    onChangeMode();
    onModalClose();
  }

  render() {
    const { onModalClose } = this.props;
    const text = i18n().preference;

    return (
      <div className="WidgetModeConfirm__container">
        <h3>{text.widgetModeConfirmTitle}</h3>
        <p className="WidgetModeConfirm__content">
          {text.widgetModeConfirmContent}
        </p>
        <div className="WidgetModeConfirm__btn-set">
          <button
            className="Btn Btn--sm Btn--gray"
            type="button"
            onClick={onModalClose}
          >
            {text.changedCancel}
          </button>
          <button
            className="Btn Btn--sm Btn--primary"
            type="button"
            onClick={this.handleChange}
          >
            {text.ok}
          </button>
        </div>
      </div>
    );
  }
}

WidgetModeConfirm.propTypes = propTypes;
WidgetModeConfirm.defaultProps = defaultProps;

export default WidgetModeConfirm;
