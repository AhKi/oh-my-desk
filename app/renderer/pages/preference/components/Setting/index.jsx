import React from 'react';
import PropTypes from 'prop-types';
import i18n from 'constants/i18n';
import ToggleButton from 'renderer/components/ToggleButton';
import WidgetModeConfirm from '../Modal/WidgetModeConfirm';
import ConfigureHotKeyContainer from '../../containers/ConfigureHotKeyContainer';
import './Setting.scss';

const propTypes = {
  defaultUserAgent: PropTypes.string,
  language: PropTypes.string,
  isOpenWidgetWhenStart: PropTypes.bool,
  isLaunchAppWhenLogin: PropTypes.bool,
  onModalOpen: PropTypes.func,
  onSetLanguageEnglish: PropTypes.func,
  onSetLanguageKorean: PropTypes.func,
  onToggleOpenWidgetWhenStart: PropTypes.func,
  onToggleOpenAppWhenLogin: PropTypes.func,
  onToggleWidgetDefaultUserAgent: PropTypes.func,
};

const defaultProps = {
  defaultUserAgent: 'DESKTOP',
  language: 'English',
  isOpenWidgetWhenStart: true,
  isLaunchAppWhenLogin: true,
  onModalOpen() {},
  onSetLanguageEnglish() {},
  onSetLanguageKorean() {},
  onToggleOpenWidgetWhenStart() {},
  onToggleOpenAppWhenLogin() {},
  onToggleWidgetDefaultUserAgent() {},
};

class Setting extends React.Component {
  constructor(props) {
    super(props);
    this.handleChangeLanguage = this.handleChangeLanguage.bind(this);
    this.handleOpenWidgetModeModal = this.handleOpenWidgetModeModal.bind(this);
  }

  handleChangeLanguage(e) {
    const language = e.target.value;
    const { onSetLanguageEnglish, onSetLanguageKorean } = this.props;

    if (language === 'English') {
      onSetLanguageEnglish();
    } else {
      onSetLanguageKorean();
    }
  }

  handleOpenWidgetModeModal() {
    const { onModalOpen, onToggleWidgetDefaultUserAgent } = this.props;

    onModalOpen(WidgetModeConfirm, {
      onChangeMode: onToggleWidgetDefaultUserAgent,
    });
  }

  render() {
    const text = i18n().preference;
    const {
      defaultUserAgent,
      language,
      isOpenWidgetWhenStart,
      isLaunchAppWhenLogin,
      onToggleOpenWidgetWhenStart,
      onToggleOpenAppWhenLogin,
    } = this.props;

    return (
      <div className="Setting">
        <div>
          <p className="Setting__title">{text.user}</p>
          <ul className="Setting__list-set">
            <li className="Setting__list">
              {text.autoStart}
              <ToggleButton
                isCheck={isLaunchAppWhenLogin}
                onToggle={onToggleOpenAppWhenLogin}
              />
            </li>
            <li className="Setting__list">
              {text.autoWidgetActive}
              <ToggleButton
                isCheck={isOpenWidgetWhenStart}
                onToggle={onToggleOpenWidgetWhenStart}
              />
            </li>
            <li className="Setting__list">
              {text.language}
              <select
                className="InputSet__select"
                value={language}
                onChange={this.handleChangeLanguage}
              >
                <option value="English">English</option>
                <option value="Korean">한국어</option>
              </select>
            </li>
          </ul>
        </div>
        <div>
          <p className="Setting__title">{text.shape}</p>
          <ul className="Setting__list-set">
            <li className="Setting__list">
              {text.defaultWidgetMode}
              <select
                className="InputSet__select"
                value={defaultUserAgent}
                onChange={this.handleOpenWidgetModeModal}
              >
                <option value="DESKTOP">{text.desktopMode}</option>
                <option value="MOBILE">{text.mobileMode}</option>
              </select>
            </li>
            <li className="Setting__list">
              {text.hotKeySearch}
              <ConfigureHotKeyContainer />
            </li>
          </ul>
        </div>
      </div>
    );
  }
}

Setting.propTypes = propTypes;
Setting.defaultProps = defaultProps;

export default Setting;
