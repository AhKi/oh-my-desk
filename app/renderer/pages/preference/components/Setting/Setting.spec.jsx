import React from 'react';
import { shallow } from 'enzyme';
import { remote } from 'electron';
import WidgetModeConfirm from '../Modal/WidgetModeConfirm';
import Setting from '.';

describe('<Setting />', () => {
  remote.getGlobal = jest.fn(() => () => JSON.stringify({ config: { language: 'English' } }));

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should match to snapshot when render default', () => {
    const wrapper = shallow(<Setting />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('test handleChangeLanguage', () => {
    const onSetLanguageEnglish = jest.fn();
    const onSetLanguageKorean = jest.fn();
    const wrapper = shallow(
      <Setting
        onSetLanguageEnglish={onSetLanguageEnglish}
        onSetLanguageKorean={onSetLanguageKorean}
      />,
    );

    it('when set English', () => {
      wrapper.instance().handleChangeLanguage({ target: { value: 'English' } });

      expect(onSetLanguageEnglish).toHaveBeenCalledTimes(1);
      expect(onSetLanguageEnglish).toHaveBeenCalledWith();
    });

    it('when set Korean', () => {
      wrapper.instance().handleChangeLanguage({ target: { value: 'Korean' } });

      expect(onSetLanguageKorean).toHaveBeenCalledTimes(1);
      expect(onSetLanguageKorean).toHaveBeenCalledWith();
    });
  });

  it('should call handleOpenWidgetModeModal', () => {
    const onModalOpen = jest.fn();
    const onToggleWidgetMode = jest.fn();
    const wrapper = shallow(
      <Setting
        onModalOpen={onModalOpen}
        onToggleWidgetMode={onToggleWidgetMode}
      />,
    );

    wrapper.instance().handleOpenWidgetModeModal();

    expect(onModalOpen).toHaveBeenCalledTimes(1);
    expect(onModalOpen).toHaveBeenCalledWith(
      WidgetModeConfirm,
      {
        onChangeMode: onToggleWidgetMode,
      },
    );
  });
});
