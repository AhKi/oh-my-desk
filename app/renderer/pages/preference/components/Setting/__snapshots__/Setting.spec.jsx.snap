// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Setting /> should match to snapshot when render default 1`] = `
<div
  className="Setting"
>
  <div>
    <p
      className="Setting__title"
    >
      User
    </p>
    <ul
      className="Setting__list-set"
    >
      <li
        className="Setting__list"
      >
        Launch oh-my-desk at login
        <ToggleButton
          checkedValue={
            <i
              className="fa fa-check"
            />
          }
          isCheck={true}
          onToggle={[Function]}
          unCheckedValue={
            <i
              className="fa fa-times"
            />
          }
          width={40}
        />
      </li>
      <li
        className="Setting__list"
      >
        Active widget when start
        <ToggleButton
          checkedValue={
            <i
              className="fa fa-check"
            />
          }
          isCheck={true}
          onToggle={[Function]}
          unCheckedValue={
            <i
              className="fa fa-times"
            />
          }
          width={40}
        />
      </li>
      <li
        className="Setting__list"
      >
        Language
        <select
          className="InputSet__select"
          onChange={[Function]}
          value="English"
        >
          <option
            value="English"
          >
            English
          </option>
          <option
            value="Korean"
          >
            한국어
          </option>
        </select>
      </li>
    </ul>
  </div>
  <div>
    <p
      className="Setting__title"
    >
      Shape
    </p>
    <ul
      className="Setting__list-set"
    >
      <li
        className="Setting__list"
      >
        Default Widget Mode
        <select
          className="InputSet__select"
          onChange={[Function]}
          value="DESKTOP"
        >
          <option
            value="DESKTOP"
          >
            Desktop Mode
          </option>
          <option
            value="MOBILE"
          >
            Mobile Mode
          </option>
        </select>
      </li>
      <li
        className="Setting__list"
      >
        Search Hotkey
        <Connect(ConfigureHotKey) />
      </li>
    </ul>
  </div>
</div>
`;
